packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -61,6 +78,37 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
         return Promise.resolve();
     }
 
+    private async _handleOnCancel(_data: OnCancelData) {
+        const lineItems = this._getLineItems();
+        const existingConsignment = this._store.getState().consignments.getConsignmentsOrThrow();
+        const { email } = this._store.getState().billingAddress.getBillingAddressOrThrow();
+        const { firstName, lastName, address1 } = existingConsignment[0].shippingAddress || {};
+        const shippingAddress = {
+            ...this._shippingData,
+            firstName: firstName !== 'Fake' ? firstName : '',
+            lastName: lastName !== 'Fake' ? lastName: '',
+            address1: address1 !== 'Fake street' ? address1 : '',
+            email: email !== 'fake@fake.fake' ? email : '',
+        };
+        const consignment = [{
+            shippingAddress,
+            lineItems,
+        }];
+        await this._store.dispatch(this._billingAddressActionCreator.updateAddress(shippingAddress));
+        if(existingConsignment && existingConsignment[0]) {
+            await this._store.dispatch(this._consignmentActionCreator.deleteConsignment(existingConsignment[0].id));
+        }
+        await this._store.dispatch(this._consignmentActionCreator.createConsignments(consignment));
```

Comment:
Should we create new consignment is `existingConsignment` is `undefined`?

Reply:
yes, we create new consignment within onCancel method to override fake data

Reply:
Correct me if I wrong:
Is existingConsignment contains fake data?
If yes, then in this case we should not create new consignment, because fake data do not exist

Reply:
fixed